apps/tests/integration/integration_test.go

```diff
@@ -92,21 +98,39 @@ type secret struct {
 }
 
 func newLabClient() (*labClient, error) {
-	clientID := os.Getenv("clientId")
-	secret := os.Getenv("clientSecret")
+	cert, privateKey, err := getCertDataFromFile()
+	if err != nil {
+		return nil, fmt.Errorf("Could not get cert data: %w", err)
+	}
 
-	cred, err := confidential.NewCredFromSecret(secret)
+	cred, err := confidential.NewCredFromCert(cert, privateKey)
 	if err != nil {
-		return nil, fmt.Errorf("could not create a cred from a secret: %w", err)
+		return nil, fmt.Errorf("Could not create a cred from the cert: %w", err)
 	}
 
-	app, err := confidential.New(microsoftAuthority, clientID, cred)
+	app, err := confidential.New(microsoftAuthority, defaultClientId, cred, confidential.WithX5C())
 	if err != nil {
 		return nil, err
 	}
 
 	return &labClient{app: app}, nil
 }
+
+func getCertDataFromFile() ([]*x509.Certificate, crypto.PrivateKey, error) {
+	data, err := os.ReadFile(pemFile)
```

Comment:
How would this work in CI (`pemFile` is constant)?

Reply:
> How would this work in CI (`pemFile` is constant)?

This PR is just to get it working locally, the next step is looking into the CI side of things. 
Potentially downloading the cert on ci machine, using openssl to convert and save to a location and then processing it in a similar way